### ISSUE TITLE
Make rules around reviewers / assignees clearer

### ISSUE THREAD

**Issue by jsms90** [foundersandcoders/master-reference#369]
**Who should students be adding as a "reviewer" and "assignee" on their PRs?**
Current facilitators from all 3 campuses makes some sense, because these people will need to be across changes, in order to communicate them to the students in their campus.

[From back in Feb:](https://github.com/foundersandcoders/master-reference/issues/155#issuecomment-282292978)
> ## _Who_ to assign our PRs to
> Re: @nelsonic 's message https://github.com/foundersandcoders/master-reference/issues/137#issuecomment-281618653
> It is not clear (at least to me), who we are supposed to be setting as reviewers/assignees of our PRs.
> + The contributing.md suggests any one of 5 people. I had assumed that assigning @des-des or @bradreeder was the most sensible, since they are the ones who made the master-reference, and it is considered a large part of their roles.
>  + Especially @bradreeder, as the curriculum developer / course coordinator who has been running all of our curriculum days, and knows everything about the feedback we have given during s-g-cs, and the curriculum development days.
>
> 1. Perhaps it would be more efficient for us to assign particular types of PR to particular individuals?
> 2. Or just assign _everyone_ on that list, so that PRs get merged as soon as any of these people are free?
> 3. Or make everyone on that list a _reviewer_ for every PR - so we have more chances for problems to be picked up on. BUT only @des-des and @bradreeder as _assignees_, so that there is someone that is actually across everything about the current state of the repo?

**Comment by des-des** [comment]
@jsms90 I think all the maintainers should be assigned.

I prefer assignment over being added for review, but that is because that is my workflow.

**Comment by jsms90** [comment]
@des-des So what is the difference between "reviewer" and "assignee"? I always took "reviewer" as someone you want to give their ok, whereas "assignee" was who you wanted/expected to actually merge?

**Comment by des-des** [comment]
Yeah for me adding someone as a reviewer is asking them to review a pull. Assigning them is passing responsibility for the pull to them review + merge.

Right now, it clearly states in the contributing file that you should assign people from the list of maintainers

**Comment by jsms90** [comment]
@des-des Yeah so in-person, all the mentors just feel really unsure about the workflow and ask me verbally what they're supposed to do. It's preventing some people from feeling confident with raising issues because they're worried that they're not doing it "properly"

Same with labels

**Comment by jsms90** [comment]
In the past, I have told mentors to assign every maintainer to every PR, as per other maintainers' suggestions. But the volume of issues and PRs obviously shot up as soon as FAC11 arrived.

I'm concerned that assigning @sofer @iteles @nelsonic @bradreeder & @des-des to every single pull request that goes through the `master-reference` is likely to become annoying and make you end up ignoring the repo altogether and/or missing anything urgent or important that you might want to be across.

What issues and PRs would be most useful for each of you to see?

At the moment, I'm assuming that the volume of activity from each maintainer is roughly indicative of their availability. Since @eliascodes and I are the ones responsible for communicating any changes to the students and mentors in our respective campuses, I'm now suggesting that mentors tag each other for review but assign only the current Course Facilitators. (I only suggest @bradreeder and @des-des if we urgently need a second reviewer.)

Personally, I'm trying to ensure that the London CF is not making larger-scale curriculum decisions by themselves, just because the London curriculum gets delivered first. So I reserve tagging / assigning @bradreeder and @des-des for issues and PRs relating to documentation or ones that have more of a curriculum-wide effect (rather than week-specific).

@bradreeder @des-des How is this working for you?

**Comment by des-des** [comment]
@jsms90 if I do not want to be assigned to pulls I can remove myself as a maintainer.

I think its fine. sofer et al can do likewise

**Comment by eliascodes** [comment]
Might the introduction of [code owners](https://github.com/blog/2392-introducing-code-owners) be something that we should take advantage of to automate some of this workflow?

**Comment by des-des** [comment]
@eliascodes :tada: YESH

**Comment by eliascodes** [comment]
I'm going to setup this up on the two workshops I'm maintainer of. If we get more master-ref maintainers agreeing then I'm happy to set it up here too.

**Comment by jsms90** [comment]
I've made a PR to the contributing guidelines, which should clarify as much as I currently understand about the process.

@des-des Sorry. I wrote this and never sent.

> I'm definitely not suggesting that anyone should remove themselves as a maintainer :worried: 
> 
> I would just really appreciate some clarity around the contribution process. In particular, so that I can communicate that to the mentors and enable them to do the work themselves.
>
> @des-des I'm more than happy to tell mentors to assign you every time. So is it the same for everyone on the list? @sofer @nelsonic @iteles @bradreeder do you all want to be assigned to every PR? If so, let's just update the contributing guidelines so that it states that and students don't need to each be asking me in person?
>
> So my next questions are:
> + Whose (if anyone's) review is _required_ before something gets merged? If it differs based on the situation, it would be useful to have some guidelines for mentors.
> + Is there a distinction between the process for PRs for each individual week and the PRs for curriculum-wide alterations?
>
> Also:
> + Who should be assigned to review PRs to each of the workshops / morning challenges (since they live in repos under `foundersandcoders` separately from `master-reference`)?
>   + Those with active maintainers 
>   + Should students or maintainers be the one to make PRs against the issues?
>    + If students are making the PRs, is the maintainer's review the only one that is required, or should there be someone else who has an overview of the whole week? 
>  + Those without active maintainers
>    + Who should be assigned to the students' PRs?

**Comment by iteles** [comment]
@sofer Sadly, I don't believe this is relevant any more? FAC may have moved in a different direction since these days?
